weave/weave_client.py

```diff
@@ -748,8 +756,14 @@ def _save_object_basic(
         return ref
 
     def _save_nested_objects(self, obj: Any, name: Optional[str] = None) -> Any:
-        if get_ref(obj) is not None:
-            return
+        if (ref := get_ref(obj)) is not None:
+            # Obj has an existing ref, check if ref is to the correct project
```

Comment:
optional: consider adding a block before this:

```
if ALLOW_MIXED_PROJECT_REFS:
   return
```

That way we can have a flag `ALLOW_MIXED_PROJECT_REFS` which helps us to a) change this logic if needed; b) isolate the feature-specific code here.

Reply:
cool, yep adding